5.drug-dependency/scripts/3.t-tests.py

```diff
@@ -0,0 +1,207 @@
+    t_test_results = ttest_ind(disease_drug_df, other_drug_df, axis=0, nan_policy='omit')
+    t_test_results_df = pd.DataFrame({
+        "drug": prism_df.columns,
+        "t_stat": t_test_results.statistic,
+        "p_value": t_test_results.pvalue
+    })
+
+    # Filter significant drugs based on p-value < 0.05
+    significant_drugs = t_test_results_df[t_test_results_df['p_value'] < 0.05]
+
+    # Perform ANOVA on these drugs
+    for drug in significant_drugs['drug'].unique():
+        # Extract drug responses for the current drug
+        disease_drug_responses = disease_drug_df[drug].dropna()
+        other_drug_responses = other_drug_df[drug].dropna()
+
+        # Perform ANOVA if both groups have sufficient data
+        if len(disease_drug_responses) > 1 and len(other_drug_responses) > 1:
+            f_statistic, p_value = f_oneway(disease_drug_responses, other_drug_responses)
+            higher_group = disease if disease_drug_responses.mean() > other_drug_responses.mean() else "Other Types"
+
+            # Store the results
+            ttest_results.append({
+                'OncotreePrimaryDisease': disease,
+                'Drug': drug,
+                'F-statistic': f_statistic,
+                'p-value': p_value,
+                'Higher in': higher_group
+            })
+
+# Convert results to DataFrame
+ttest_results_df = pd.DataFrame(ttest_results)
+
+# Apply a significance threshold (e.g., p < 0.05)
+significant_ttest_results_df = ttest_results_df[ttest_results_df['p-value'] < 0.05]
+
+# Display the top 50 significant results based on F-statistic
+significant_ttest_results_df.sort_values(by='F-statistic', key=abs, ascending=False).head(50)
+
+
+# In[7]:
+
+
+# Assuming 'drug_column_name' is the column in prism_trt_df that matches the 'drug' column in correlation_df
+prism_trt_df_filtered = prism_trt_df[['column_name', 'name', 'moa', 'target']]
+
+# Merge correlation_df with prism_trt_df based on the 'drug' column in correlation_df and the matching column in prism_trt_df
+merged_df = pd.merge(significant_ttest_results_df, prism_trt_df_filtered, how='left', left_on='Drug', right_on='column_name')
+
+# Drop the redundant drug_column_name column after the merge if needed
+merged_df = merged_df.drop(columns=['column_name'])
+
+# Save results to a CSV file
+merged_df.to_csv("../5.drug-dependency/results/drug_diff_results.csv", index=False)
+
+
+# In[8]:
+
+
+# Assuming merged_df and correlation_df are already defined
+correlation_df_filtered = correlation_df[['drug', 'latent_dimension', 'correlation', 'indication','phase']]
+
+# Get unique values from "Higher in" column and exclude "Other types"
+higher_in_values = merged_df["Higher in"].unique()
+higher_in_values = [value for value in higher_in_values if value != "Other Types"]
+
+pdf_path = pathlib.Path('../5.drug-dependency/results/Individual_Drug_Analysis.pdf').resolve()
+
+# Create a PDF file to save the plots
+with PdfPages(pdf_path) as pdf:
```

Comment:
Same with comment about the PDF, confirm that this is outputting a PDF that is in proper format. Currently, for example, there is a title in the middle of the rows. This is a cool way to output the info!

Reply:
Yep, there should be one page of table for a certain cancer type and then a page of unique drugs, then back to table for a different cancer type!